src/toil/provisioners/gceProvisioner.py

```diff
@@ -351,13 +355,14 @@ def launchCluster(self, leaderNodeType, leaderSpotBid, nodeTypes, preemptableNod
         #TODO: get subnetID
         #self.subnetID = leader.subnet_id
 
-        if not self._copySshKeys(leader.public_ips[0], keyName) \
-           or not self._waitForNode(leader.public_ips[0], 'toil_leader'):
-            raise RuntimeError("Failed to start leader")
+        if (not self._waitForNode(leader.public_ips[0], 'toil_leader')
+            or not self._copySshKeys(leader.public_ips[0], keyName)
+            or not self._injectFile(leader.public_ips[0], self.credentialsPath, GoogleJobStore.nodeServiceAccountJson,
+                                    'toil_leader')
+            or (self.botoPath and
+                not self._injectFile(leader.public_ips[0], self.botoPath, self.nodeBotoPath, 'toil_leader'))):
 
-        if self.botoPath:
-            self._rsyncNode(leader.public_ips[0], [self.botoPath, ':' + nodeBotoPath],
-                            applianceName='toil_leader')
+            raise RuntimeError("Failed to start leader")
```

Comment:
I recall making this comment before, but I think each of these conditions should throw a different error message to relate to the user why it failed.